workbench/tests/jest.overrides.js

```diff
@@ -1,4 +1,27 @@
-const crypto = require('crypto');
+import crypto from 'crypto';
+import fetch from 'node-fetch';
+
+import api from '../src/preload/api';
+
+// debug logging is a bit noisy, not so useful during tests.
+if (!process.env.ELECTRON_LOG_LEVEL) {
+  process.env.ELECTRON_LOG_LEVEL = 'info';
+}
+
+// Test for a jsdom env (as opposed to node), which means renderer tests.
+if (global.window) {
+  // mock the work of preload.js here:
+  global.window.Workbench = api;
+
+  // mock out the global gettext function - avoid setting up translation
+  global.window._ = (x) => x;
+
+  // jsdom does not implement these window APIs:
+  global.window.crypto = {
+    getRandomValues: () => [crypto.randomBytes(4).toString('hex')],
```

Comment:
Yes good question, the dueling `crypto` api confusion continues...

Here we're setting up the APIs we need for the renderer tests, which run in a `jsdom` environment, which is _almost_ like the browser, but lacks some built-in browser APIs, like `window.crypto` and `window.fetch`. So we're patching in `crypto` into jsdom's `global.window`, but since the runtime for this setup script is node...well we only have the node module `crypto` available to generate some random data.

Reply:
Ahhh, got it, thanks.